File: server/bcephal-reco/src/main/java/com/moriset/bcephal/reconciliation/domain/AutoRecoCommonDimension.java
/**
 * 
 */
package com.moriset.bcephal.reconciliation.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.Persistent;
import com.moriset.bcephal.domain.dimension.DimensionType;
import com.moriset.bcephal.domain.filters.PeriodValue;

import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Entity(name = "AutoRecoCommonDimension")
@Table(name = "BCP_AUTO_RECO_COMMON_DIMENSION")
@Data
@EqualsAndHashCode(callSuper = false)
public class AutoRecoCommonDimension extends Persistent {
	
	private static final long serialVersionUID = 3234923830535725220L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "auto_reco_common_dimension_seq")
	@SequenceGenerator(name = "auto_reco_common_dimension_seq", sequenceName = "auto_reco_common_dimension_seq", initialValue = 1,  allocationSize = 1)
	private Long id;
	
	@JsonIgnore
	private Long recoId;
		
	@Enumerated(EnumType.STRING)
	private DimensionType dimensionType;
	
	private Long dimensionId;
	
	private String dimensionName;
			
	@Enumerated(EnumType.STRING) 
	private AutoRecoPeriodCondition periodCondition;
	
	@Embedded	
	private PeriodValue dateValue;

	
	@JsonIgnore
	@Override
	public Persistent copy() {
		return null;
	}

		
	
}

File: client/bcephal-client-model/Reconciliation/AutoRecoCommonDimension.cs
using Bcephal.Models.Base;
using Bcephal.Models.Filters;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Reconciliation
{
    public class AutoRecoCommonDimension : Persistent
    {

		public DimensionType DimensionType { get; set; }

		public long? DimensionId { get; set; }

		public string DimensionName { get; set; }

		public string PeriodCondition { get; set; }
		[JsonIgnore]
		public AutoRecoPeriodCondition PeriodConditions
		{
			get
			{
				return string.IsNullOrEmpty(PeriodCondition) ? null : AutoRecoPeriodCondition.GetByCode(PeriodCondition);
			}
			set
			{
				this.PeriodCondition = value != null ? value.getCode() : null;
			}
		}

		public PeriodValue DateValue { get; set; }

	}
}
